src/Discord.Net.Commands/Info/CommandInfo.cs

```diff
@@ -63,6 +63,9 @@ internal CommandInfo(CommandBuilder builder, ModuleInfo module, CommandService s
             _action = builder.Callback;
         }
 
+        public TPrecondition GetPrecondition<TPrecondition>() where TPrecondition : PreconditionAttribute =>
```

Comment:
This should return an `IEnumerable` of preconditions instead of using `FirstOrDefault`. We can't assume the user wants only the first result.